### DISCUSSION TITLE
User Table  - Bug?

### DISCUSSION THREAD

**Discussion by SwiPowE** [Budibase/budibase#3459] [Help]
Hi to all

I am starting with budibase and want to develop an event registration process. I really like it and see lots of potential..keep on with the great work. 

Which data is appearing in the user tables? Why can I add users in this table through an app, but they won't appear in the portal? 
And: Why can't these app created users not be deleted or updated? E-Mail should be unique in the users table, why can I add several lines with the same email? Is there a way to specify users unique?

Moreover there are some strange filter results. 

I have 3 users in the portal:
![image](https://user-images.githubusercontent.com/93815489/142754757-89b7ea28-4961-4752-be91-ae062988d14a.png)
 
In the default Users table there is appearing just of those users and the ones I created via app.
 
![image](https://user-images.githubusercontent.com/93815489/142754819-324330de-0bcb-485d-bac2-50f38c9758f0.png)

When I create a view on the [Users] Table all users of my tenant appear. 

![image](https://user-images.githubusercontent.com/93815489/142754848-309986ad-68bb-4455-9ed9-936de506026c.png)

I don't understand the Users Table in the app. I am able to add users, but email is not the key..is there a way to have the email unique?

Thank you and best regards from switzerland!

Keep on!

SwiPowE aka Bernd

**Comment by mjashanks** [comment]
Hey Bernd. Thank you for writing this up. This is indeed a bug. You should not be able to add users via the app. The portal is the only place that users should be added/edited.

Thanks - I'll get this logged as a bug.

**Reply by gvlx** [reply]
Also: users added through the portal do not appear on the App's `Users` table.

**Reply by gvlx** [reply]
To make it clear: I cannot add users on the Data pane (there is no `Create Row` button), but I can create a form to add new users on to this table.

**Reply by mike12345567** [reply]
Hi @gvlx - thanks for this info, in the next release users added through the portal will appear in the users tables of all apps they have access to (they aren't just a public role).

We will also need to look at disabling the ability to create a form which adds new users in an app as these aren't platform users and won't actually be valid/able to login.
